src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -50,14 +50,20 @@
   private static final String KEY_SCHEMA_ID = "key_schema_id";
   static final String HEADERS = "headers";
 
-  public static final SimpleDateFormat ISO_DATE_FORMAT =
-      new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-  public static final SimpleDateFormat TIME_FORMAT = new SimpleDateFormat("HH:mm:ss.SSSZ");
-  static final int MAX_SNOWFLAKE_NUMBER_PRECISION = 38;
+  // For each task, we require a separate instance of SimpleDataFormat, since they are not
+  // inherently thread safe
+  private static final ThreadLocal<SimpleDateFormat> ISO_DATE_FORMAT =
+      ThreadLocal.withInitial(
+          () -> {
+            SimpleDateFormat simpleDateFormat =
+                new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+            simpleDateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+            return simpleDateFormat;
+          });
 
-  static {
-    ISO_DATE_FORMAT.setTimeZone(TimeZone.getTimeZone("UTC"));
-  }
+  public static final ThreadLocal<SimpleDateFormat> TIME_FORMAT =
+      ThreadLocal.withInitial(() -> new SimpleDateFormat("HH:mm:ss.SSSZ"));
```

Comment:
Time_format doesnt have timezone set. Should we do this behavior change and be uniform?

Reply:
CC: @sfc-gh-zli

Reply:
Can you explain the customer facing impact of this behavior change?

Reply:
My understanding is that, this `time_format` has the timezone which is dependent on worker node timezone which usually is UTC but it is not explicit.

Reply:
(We don't have to do this behavior change, I am just starting this conversation here as it feels more related)